lib/internal/bootstrap/loaders.js

```diff
@@ -157,15 +156,19 @@ function NativeModule(id) {
   this.exportKeys = undefined;
   this.loaded = false;
   this.loading = false;
-  if (id === loaderId) {
+  this.canBeRequiredByUsers = !id.startsWith('internal/');
+}
+
+// To be called during pre-execution when --expose-internals is on.
+// Enables the user-land module loader to access internal modules.
+NativeModule.exposeInternals = function() {
+  for (const [id, mod] of NativeModule.map) {
     // Do not expose this to user land even with --expose-internals.
-    this.canBeRequiredByUsers = false;
-  } else if (id.startsWith('internal/')) {
-    this.canBeRequiredByUsers = exposeInternals;
-  } else {
-    this.canBeRequiredByUsers = true;
+    if (id.startsWith('internal/') && id !== loaderId) {
+      mod.canBeRequiredByUsers = true;
```

Comment:
`mod.canBeRequiredByUsers = (id.startsWith('internal/') && id !== loaderId);`? Any reason to set it only conditionally?

(I'd swap the clauses for efficiency but that aside.)